From cc1e7d0a306ad2f4efc4750866fc16454e6899cc Mon Sep 17 00:00:00 2001
From: Fabian Orccon <cfoch.fabian@gmail.com>
Date: Fri, 22 May 2026 14:20:10 +0200
Subject: [PATCH 12/24] gmain: emscripten: Use 0 timeout

As of the date non-zero timeouts are not supported on poll()

Ported with the G_PLATFORM_WASM guard in place of G_WITH_EMSCRIPTEN,
and against the microsecond timeout variable the iteration uses now.

(adapted from commit 990ffd85fe4167bf3a28a28c6f85a25287ca79ad)
---
 glib/gmain.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/glib/gmain.c b/glib/gmain.c
index e3c72d37e..60d949541 100644
--- a/glib/gmain.c
+++ b/glib/gmain.c
@@ -4507,7 +4507,9 @@ g_main_context_iterate_unlocked (GMainContext *context,
       context->cached_poll_array = fds = g_new (GPollFD, nfds);
     }
 
+#ifndef G_PLATFORM_WASM
   if (!block)
+#endif
     timeout_usec = 0;
 
   g_main_context_poll_unlocked (context, timeout_usec, max_priority, fds, nfds);
